packages/router/src/__tests__/router.test.tsx

```diff
@@ -509,6 +509,38 @@ test('params should never be an empty object in Set', async (done) => {
   render(<TestRouter />)
 })
 
+test.only('params should never be an empty object in Set', async () => {
+  const ParamPage = () => {
+    const { documentId } = useParams()
+    return 'documentId: ' + documentId
+  }
+
+  const SetWithUseParams = ({ children }) => {
+    const params = useParams()
+    // 1st run: { documentId: '1' }
+    // 2nd run: {}
```

Comment:
I'm not sure how this is possible, since the Set's children's path isn't matched and `null` is returned instead.

Maybe this has something to do with the way we're wrapping the sets into each other. 🤷‍♂️

Reply:
I am not 100% sure but I believe this is cause by
https://github.com/redwoodjs/redwood/blob/eb0e0a9917b82b7b1af76a596824e7c0e15c7bfd/packages/router/src/page-loader.tsx#L131